app/presenters/bid_status_flash_factory.rb

```diff
@@ -0,0 +1,175 @@
+class BidStatusFlashFactory
+  attr_reader :auction, :flash, :user
+
+  def initialize(auction:, flash:, user:)
+    @auction = auction
+    @flash = flash
+    @user = user
+  end
+
+  def create
+    if available?
+      available_flash
+    else
+      over_flash
+    end
+  end
+
+  private
+
+  def available_flash
+    if user_is_bidder? && flash['bid']
+      BidSubmitted.new
+    elsif user_is_winning_bidder?
+      AvailableUserIsWinningBidder.new
+    else # user is bidder
+      AvailableUserIsBidder.new(bid_amount: lowest_user_bid.try(:amount))
+    end
+  end
+
+  def over_flash
+    if user_is_winning_bidder?
+      OverUserIsWinner.new
+    elsif user_is_bidder?
+      OverUserIsBidder.new
+    elsif auction.bids.any?
+      OverWithBids.new
+    else
+      OverNoBids.new
+    end
+  end
+
+  def over?
+    auction_status.over?
+  end
+
+  def available?
+    auction_status.available?
+  end
+
+  def auction_status
+    AuctionStatus.new(auction)
+  end
+
+  def user_is_winning_bidder?
+    user_bids.any? && lowest_user_bid == auction.lowest_bid
+  end
+
+  def user_is_bidder?
+    user_bids.any?
+  end
+
+  def lowest_user_bid
+    user_bids.order(amount: :asc).first
+  end
+
+  def user_bids
+    auction.bids.where(bidder: user)
+  end
+end
+
+class OverWithBids
```

Comment:
it is working here since we are only referencing the classes within this file -- what do you think?